tests/io/dask/delayed/test_index.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+import pickle
+
+import dask
+import pytest
+
+from kartothek.io.dask.delayed import build_dataset_indices__delayed
+from kartothek.io.testing.index import *  # noqa: F4
+
+
+def _build_indices(*args, **kwargs):
+    tasks = build_dataset_indices__delayed(*args, **kwargs)
+    s = pickle.dumps(tasks, pickle.HIGHEST_PROTOCOL)
```

Comment:
why pickle stuff? We're using distributed for our test setup and there should be no need to test serializability _manually_

Reply:
distributed uses "pickle with cloudpickle" fallback. However, cloudpickle (the fallback) is massively more inefficient. I'm OK when an average users writes bags/graph that need to be cloudpickled (e.g. due to nested classes/functions or lambdas) but I am expecting ktk to be efficient. These roundtrip tests are used for other dask tasks as well.

Reply:
Fair enough. Can you put a comment in there? Maybe we can add a test like this to all pipelines (not this PR)

Reply:
Let's do the generic pipeline stuff later, since this would require some more glue code and some testing that it works. I have some ideas on how to wire this up, but this takes some time.